pandas/core/window/rolling.py

```diff
@@ -538,18 +539,22 @@ class BaseWindowGroupby(BaseWindow):
     Provide the groupby windowing facilities.
     """
 
+    _grouper: BaseGrouper
+    _as_index: bool
     _attributes = ["_grouper"]
 
     def __init__(
         self,
         obj: FrameOrSeries,
         *args,
-        _grouper=None,
-        _as_index=True,
+        _grouper: BaseGrouper,
+        _as_index: bool = True,
         **kwargs,
     ):
-        if _grouper is None:
-            raise ValueError("Must pass a Grouper object.")
+        from pandas.core.groupby.ops import BaseGrouper
```

Comment:
you reallly need this import (as this is imported above right?)

Reply:
its imported above inside a `if TYPE_CHECKING`; importing it at the module level raises bc of circularity

Reply:
i c, ok